### PR TITLE
ARROW-1714: [Python] Fix invalid serialization/deserialization None name Series

### PR DESCRIPTION
**Pull request by Licht-T**
This closes [ARROW-1714](https://issues.apache.org/jira/projects/ARROW/issues/ARROW-1714).

### CODE REVIEW

File: python/pyarrow/pandas_compat.py [hunk 1]
```diff
@@ -450,6 +455,18 @@ def table_to_blockmanager(options, table, memory_pool, nthreads=1):
         index = pd.RangeIndex(row_count)
 
     column_strings = [x.name for x in block_table.itercolumns()]
+    if 'columns_metadata' in locals() and columns_metadata is not None:
```
Comment (wesm) [inline python/pyarrow/pandas_compat.py]: I'm not a fan of using `locals()` though I don't know exactly why. I'm going to remove this and make `columns_metadata` always None even if there is no pandas metadata

**Review by wesm — APPROVED** [review]
+1, I made a minor comment which I'll quickly fix, and then will merge once the build runs

**Comment by wesm** [comment]
@cpcloud do you know what's going on here? Seems related to https://github.com/apache/arrow/commit/b221a2c7a08f4f709c3a4eb4b2d68dae16c82da8

**Comment by wesm** [comment]
Think this just needed to be rebased

**Comment by wesm** [comment]
Rebased again. These tests are passing locally for me now

**Comment by wesm** [comment]
This seems to only fail with pandas 0.21.0, @Licht-T can you take a look?

**Comment by Licht-T** [comment]
Thanks @wesm , I'll check.

**Comment by Licht-T** [comment]
@wesm I found the reason why it won't work.
https://github.com/apache/arrow/commit/b221a2c7a08f4f709c3a4eb4b2d68dae16c82da8 does not treat when  some `labels` are `-1`.

```python
>>> pd.MultiIndex.from_tuples([(1,)])
MultiIndex(levels=[[1]],
           labels=[[0]])
>>> pd.MultiIndex.from_tuples([(None,)])
MultiIndex(levels=[[]],
           labels=[[-1]])
```

**Comment by wesm** [comment]
OK, let me try to fix it

**Comment by wesm** [comment]
+1, will merge once the build passes

**Review by xhochy — APPROVED** [review]

**Pull request closed without merging**